src/azure-cli/azure/cli/command_modules/role/tests/latest/test_role.py

```diff
@@ -720,15 +722,19 @@ def test_role_assignment_no_graph(self, resource_group):
             self.cmd('role assignment create '
                      '--assignee-object-id {uami_object_id} --assignee-principal-type ServicePrincipal '
                      '--role {role_reader_guid} --scope {rg_id}')
-            # Verify '--fill-principal-name false' skips the Graph query for filling principalName.
-            self.cmd('role assignment list --scope {rg_id} --fill-principal-name false '
-                     '--fill-role-definition-name false',
+            # --assignee-object-id bypasses Graph query for resolving assignee object ID
+            # '--fill-role-definition-name false' bypasses role definitions query for filling roleDefinitionName
+            # '--fill-principal-name false' bypasses Graph query for filling principalName
+            self.cmd('role assignment list --all --assignee-object-id {uami_object_id} '
+                     '--fill-role-definition-name false --fill-principal-name false',
```

Comment:
Using `--all --assignee-object-id` reduces the recording file size (https://github.com/Azure/azure-cli/pull/31152#discussion_r2026378172).